library/src/main/java/com/datatorrent/lib/io/fs/HDFSFileSplitter.java

```diff
@@ -0,0 +1,180 @@
+
+    Path path = new Path(fileInfo.getFilePath());
+    FileStatus status = getFileStatus(path);
+    if (fileInfo.getDirectoryPath() == null) { // Direct filename is given as input.
+      hdfsFileMetaData.setRelativePath(status.getPath().getName());
+    } else {
+      String relativePath = getRelativePathWithFolderName(fileInfo);
+      hdfsFileMetaData.setRelativePath(relativePath);
+    }
+    return hdfsFileMetaData;
+  }
+
+  /*
+   * As folder name was given to input for copy, prefix folder name to the sub items to copy.
+   */
+  private String getRelativePathWithFolderName(FileInfo fileInfo)
+  {
+    String parentDir = new Path(fileInfo.getDirectoryPath()).getName();
+    return parentDir + File.separator + fileInfo.getRelativeFilePath();
+  }
+
+  @Override
+  protected HDFSBlockMetadata createBlockMetadata(FileMetadata fileMetadata)
+  {
+    HDFSBlockMetadata blockMetadta = new HDFSBlockMetadata(fileMetadata.getFilePath());
+    blockMetadta.setReadBlockInSequence(sequencialFileRead);
+    return blockMetadta;
+  }
+
+  @Override
+  protected HDFSBlockMetadata buildBlockMetadata(long pos, long lengthOfFileInBlock, int blockNumber, FileMetadata fileMetadata, boolean isLast)
+  {
+    FileBlockMetadata metadata = super.buildBlockMetadata(pos, lengthOfFileInBlock, blockNumber, fileMetadata, isLast);
+    HDFSBlockMetadata blockMetadata = (HDFSBlockMetadata)metadata;
+    return blockMetadata;
+  }
+
+  public boolean isSequencialFileRead()
+  {
+    return sequencialFileRead;
+  }
+
+  public void setSequencialFileRead(boolean sequencialFileRead)
+  {
+    this.sequencialFileRead = sequencialFileRead;
+  }
+
+  public static class HDFSScanner extends TimeBasedDirectoryScanner
+  {
+    protected static final String HDFS_TEMP_FILE = ".*._COPYING_";
+    protected static final String UNSUPPORTED_CHARACTOR = ":";
+    private transient Pattern ignoreRegex;
+
+    @Override
+    public void setup(OperatorContext context)
+    {
+      super.setup(context);
+      ignoreRegex = Pattern.compile(HDFS_TEMP_FILE);
+    }
+
+    @Override
+    protected boolean acceptFile(String filePathStr)
+    {
+      boolean accepted = super.acceptFile(filePathStr);
+      if (containsUnsupportedCharacters(filePathStr) || isTempFile(filePathStr)) {
+        return false;
+      }
+      return accepted;
+    }
+
+    private boolean isTempFile(String filePathStr)
+    {
+      String fileName = new Path(filePathStr).getName();
+      if (ignoreRegex != null) {
+        Matcher matcher = ignoreRegex.matcher(fileName);
+        if (matcher.matches()) {
+          return true;
+        }
+      }
+      return false;
+    }
+
+    private boolean containsUnsupportedCharacters(String filePathStr)
+    {
+      return new Path(filePathStr).toUri().getPath().contains(UNSUPPORTED_CHARACTOR);
+    }
+  }
+
+  public static class HDFSFileMetaData extends FileMetadata
+  {
+    private String relativePath;
+
+    protected HDFSFileMetaData()
```

Comment:
Yes, definitely, I was just concerned if that would be accepted in generic class or not.